src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestGetCallGen.generated.txt

```diff
@@ -0,0 +1,89 @@
+        {
+            throw new global::System.ArgumentNullException(nameof(configuration));
+        }
+        if (typeof(T) == typeof(Program.MyClass))
+        {
+            Microsoft.Extensions.Configuration.IConfigurationSection? section = configuration as Microsoft.Extensions.Configuration.IConfigurationSection;
+            if (section?.Value is null && !configuration.GetChildren().Any())
+            {
+                return default;
+            }
+            Program.MyClass obj = new Program.MyClass();
+            BindCore(configuration, ref obj);
+            return (T)(object)obj;
+        }
+
+        throw new global::System.NotSupportedException($"Unable to bind to type '{typeof(T)}': 'Generator parser did not detect the type as input'");
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+        if (configuration.GetSection("MyString").Value is string stringValue1) { obj.MyString = stringValue1; }
+        if (configuration.GetSection("MyInt").Value is string stringValue2) { obj.MyInt = int.Parse(stringValue2); }
+        System.Collections.Generic.List<int> temp3 = obj.MyList;
+        temp3 ??= new System.Collections.Generic.List<int>();
+        BindCore(configuration.GetSection("MyList"), ref temp3);
+        obj.MyList = temp3;
+        System.Collections.Generic.Dictionary<string, string> temp4 = obj.MyDictionary;
+        temp4 ??= new System.Collections.Generic.Dictionary<string, string>();
+        BindCore(configuration.GetSection("MyDictionary"), ref temp4);
+        obj.MyDictionary = temp4;
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.List<int> obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+        int element;
+        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
+        {
+            if (section?.Value is string stringValue5)
+            {
+                element = int.Parse(stringValue5);
+                obj.Add(element);
+            }
+        }
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.Dictionary<string, string> obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+        string key;
+        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
+        {
+            if (section?.Key is string stringValue6)
+            {
+                key = stringValue6;
+                if (key is not null)
```

Comment:
We shouldn't need these null checks.

1. `section` is non-nullable (GetChildren() returns an enumerable of non-nullable `IConfigurationSection` instances.
2. `section.Key` is non-nullable.